File: 2256.minimum-average-difference/solution.py
class Solution:
    def minimumAverageDifference(self, nums: List[int]) -> int:
        N = len(nums)
        if (N == 1):
            return 0

        totalL = nums[0]
        totalR = sum(nums) - totalL

        cntL = 1
        cntR = N - 1
        minIdx = 0
        minDiff = abs(totalL - totalR // cntR)

        for i in range(1, N):
            cntL += 1
            cntR -= 1
            totalL += nums[i]
            totalR -= nums[i]

            if (cntR == 0):
                diff = abs(totalL//cntL)
            else:
                diff = abs(abs(totalL // cntL) - abs(totalR // cntR))
            # print(f"{totalL}//{cntL} - {totalR}//{cntR} = {diff}")

            if diff < minDiff:
                minDiff = diff
                minIdx = i
        return minIdx

File: 2256.minimum-average-difference/solution.kt
import kotlin.math.*
class Solution {
    fun minimumAverageDifference(nums: IntArray): Int {
        var rightS = 0L
        val numsL = nums.map {
            val itLong = it.toLong()
            rightS += itLong
            itLong
        }

        val N = numsL.size
        var leftS = 0L
        var leftCnt = 0
        var rightCnt = N

        var minIdx = 0
        var minDiff = Long.MAX_VALUE

        for (i in 0 until nums.size) {
            leftCnt += 1
            rightCnt -= 1
            leftS += numsL[i]
            rightS -= numsL[i]

            var diff = abs(when(rightCnt) {
                0 -> leftS / leftCnt
                else -> abs(leftS / leftCnt) - abs(rightS / rightCnt)
            })

            if (diff < minDiff) {
                minDiff = diff
                minIdx = i
            }
        }
        return minIdx
    }
}
